api/proto/teleport/legacy/client/proto/authservice.proto

```diff
@@ -493,12 +493,12 @@ message Features {
   bool SAML = 5 [(gogoproto.jsontag) = "saml"];
   // AccessControls enables FIPS access controls
   bool AccessControls = 6 [(gogoproto.jsontag) = "access_controls"];
-  // AdvancedAccessWorkflows enables advanced access workflows
-  // Deprecated: Use the [AccessRequests.Enabled] field instead.
-  bool AdvancedAccessWorkflows = 7 [
-    (gogoproto.jsontag) = "advanced_access_workflows",
-    deprecated = true
-  ];
+  // AdvancedAccessWorkflows enables advanced access workflows.
+  //
+  // This field is now a legacy flag with the introduction of Enterprise Usage Based (EUB)
+  // product. It will still be used for backwards compatibility for legacy feature support
+  // where existing licenses before EUB should still have the same support as before.
+  bool AdvancedAccessWorkflows = 7 [(gogoproto.jsontag) = "advanced_access_workflows"];
```

Comment:
I was affected by the previous PR but can't you just still mark the field as deprecated and during init you run

```go
if features.AdvancedAccessWorkflows {
features.AccessRequests.Enabled=true

}
```

since from what I understood, `AdvancedAccessWorkflows` only gated the access request feature.

Then, you can adapt the code and eventually get rid of `AdvancedAccessWorkflows` field

Reply:
If I understand correctly, to migrate to another flag such as `AccessRequests.Enabled`, we would need to:

1. Introduce Tiago's fix for legacy self-hosted licenses (since we can't reissue licenses for everyone)
2. Update products in Sales Center to have the `AccessRequests` feature enabled. <- This IMO is the right way to solve this for instances where the feature source is cloud, rather than hacks in the code.

Reply:
sorry i wasnt' sure what the suggestion was.

i reverted the PR that did the name change because i didn't want to deprecate a legacy field `AdvancedAccessWorkflows` only to introduce another legacy field `AccessRequests.Enabled`, because `features.IdentityGovernance` replaces it. 

also i noticed, that `AdvancedAccessWorkflows` seems to be redundant with `isBuildEnterprise` because salescenter Team plan, sets `AccessRquests = true`, and for on-prem and enterprise cloud, we hard code it to true. So for all enterprise editions currently, the AR is enabled.

Reply:
I mean, if the change makes sense from product perspective (I think it does but I have no context) you should keep it

Reply:
@tigrato for now, i am going to leave it as is (keep the revert and keep `AdvancedAccessWorkflows`), here is a [slack chat](https://gravitational.slack.com/archives/C05M9TC5WKD/p1700706130808939) that pretty much confirmed it will become obselete